packages/typespec-azure-core/src/decorators.ts

```diff
@@ -268,15 +267,11 @@ export function getItems(program: Program, entity: Type): boolean | undefined {
   return program.stateMap(AzureCoreStateKeys.items).get(entity);
 }
 
-export const $nextLink: NextLinkDecorator = (context: DecoratorContext, entity: ModelProperty) => {
-  context.program.stateMap(AzureCoreStateKeys.nextLink).set(entity, true);
-};
-
 /**
  * Returns `true` if the property is marked with `@nextLink`.
  */
-export function getNextLink(program: Program, entity: Type): boolean | undefined {
-  return program.stateMap(AzureCoreStateKeys.nextLink).get(entity);
+export function getNextLink(program: Program, entity: ModelProperty): boolean {
```

Comment:
Why change the signature?  We would like to not break the behavior of this function, for those using it